client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-generic-form.tsx

```diff
@@ -0,0 +1,868 @@
+
+import './style.scss';
+
+export const GENERIC_EMAIL_FORM_EMAIL_FIELD = 'email';
+export const GENERIC_EMAIL_FORM_FULL_NAME_FIELD = 'fullName';
+export const GENERIC_EMAIL_FORM_FIRST_NAME_FIELD = 'firstName';
+export const GENERIC_EMAIL_FORM_LAST_NAME_FIELD = 'lastName';
+export const GENERIC_EMAIL_FORM_IS_ADMIN_FIELD = 'isAdmin';
+export const GENERIC_EMAIL_FORM_ALTERNATIVE_EMAIL_FIELD = 'alternativeEmail';
+
+const requiredField = ( { value, error }: GenericNewUserField ): GenericNewUserField => ( {
+	value,
+	error:
+		! error && ( ! value || '' === value.trim() ) ? translate( 'This field is required.' ) : error,
+} );
+
+/*
+ * Adds a new error if the mailbox has no errors and the email address is invalid.
+ */
+const validateEmail = ( {
+	value: alternativeEmail,
+	error: alternativeEmailError,
+}: GenericNewUserField ): GenericNewUserField => ( {
+	value: alternativeEmail,
+	error:
+		! alternativeEmailError && ! emailValidator.validate( alternativeEmail )
+			? translate( 'Please provide a valid email address.' )
+			: alternativeEmailError,
+} );
+
+/*
+ * Adds a new error if the mailbox has no errors and the email address is invalid.
+ */
+const validateOverallEmail = (
+	{ value: mailBox, error: mailBoxError }: GenericNewUserField,
+	{ value: domain }: GenericNewUserField
+): GenericNewUserField => ( {
+	value: mailBox,
+	error:
+		! mailBoxError && ! emailValidator.validate( `${ mailBox }@${ domain }` )
+			? translate( 'Please provide a valid email address.' )
+			: mailBoxError,
+} );
+
+/**
+ * Retrieves the specified user after applying a callback to all of its fields.
+ *
+ * @param {object} user - user with a list of fields
+ * @param {Array} optionalFields - list of optional fields (not required fields)
+ * @param {Function} callback - function to call for each field
+ */
+const mapFieldValues = (
+	user: GenericNewUser,
+	optionalFields: string[],
+	callback: any
+): GenericNewUser =>
+	mapValues( user, ( fieldValue, fieldName ) => {
+		if ( 'uuid' === fieldName || optionalFields.some( ( field: string ) => field === fieldName ) ) {
+			return fieldValue;
+		}
+
+		return callback( fieldValue, fieldName, user );
+	} );
+
+/*
+ * Adds a new error to the specified email field if it has no errors and contains invalid characters.
+ */
+const validEmailCharacterField = ( {
+	value,
+	error,
+}: GenericNewUserField ): GenericNewUserField => ( {
+	value,
+	error:
+		! error && ! /^[0-9a-z_'-](\.?[0-9a-z_'-])*$/i.test( value )
+			? translate(
+					'Only numbers, letters, dashes, underscores, apostrophes and periods are allowed.'
+			  )
+			: error,
+} );
+
+/*
+ * Adds a new error to the specified field if it has no errors and is more than sixty characters.
+ */
+const sixtyCharacterField = ( { value, error }: GenericNewUserField ): GenericNewUserField => ( {
+	value,
+	error:
+		! error && 60 < value.length
+			? translate( "This field can't be longer than %s characters.", {
+					args: '60',
```

Comment:
Should this not be `%d`? We should add a comment to make it clear that the number is 60 (i.e. quite large).
```suggestion
			? translate( "This field can't be longer than %d characters.", {
					args: 60,
					comment: '%d is a number of characters, normally 60',
```